### PR TITLE
Revise developer documentation

### PR DESCRIPTION
**Pull request by t-b**
Remove CONTRIBUTE.md as it is superfluous.

Rework INSTALL.md with the follwoing goals:
- Describe all CMake variables in a easy-to-read manner
- Explain how cross-compiling tango works
- Explain how to compile the tango dependencies

@bourtemb @mliszcz Sorry for rushing this now.

But I would need that for tuesday. I've also given you write access to the forked repo. So for bigger changes please push them as fixup commits. This is way more convenient than github comments.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -1,96 +1,190 @@
...
+
+## tango-idl
+
+```bash
+git clone https://github.com/tango-controls/tango-idl
+cd tango-idl
+mkdir build
+cd build
+cmake ..
+make [-j NUMBER_OF_CPUS]
```
Comment (mliszcz) [inline INSTALL.md]: You don't need this step, 'all' target is empty in tango-idl, and anyway '-j' won't speed up anything in such simple package.

BTW I noticed that you can't run these steps without c++ compiler being installed, which makes no sense. But maybe this requirement comes from CMake.

File: configure/toolchain-i686.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
+set(CMAKE_CROSSCOMPILING TRUE)
```
Comment (mliszcz) [inline configure/toolchain-i686.cmake:1]: IMO this file should be merged as part of #792 or at least as a separate commit in this PR, just to keep the history clear.

**Review by mliszcz — DISMISSED** [review]
Looks good to me, thanks for cleaning this file up!

**Review by bourtemb — COMMENTED** [review]
The old CONTRIBUTE.md was still in Draft mode since it was created and not updated since then.
I think it could be useful to keep a CONTRIBUTE.md file because it integrates well with Github interface.
See https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/setting-guidelines-for-repository-contributors
Maybe we should update CONTRIBUTE.md and simply put a link to the wiki page as in The README.md file?

**Comment by t-b** [comment]
@bourtemb The wiki page is pretty ugly as well. But I can link to it from CONTRIBUTE.md if you think this is useful.

**Comment by bourtemb** [comment]
> @bourtemb The wiki page is pretty ugly as well. But I can link to it from CONTRIBUTE.md if you think this is useful.

I think it is useful if the wiki page is up to date. Please feel free to update it if you think it can be improved.
I think the original idea behind creating the wiki page was to make it easy to write and update.
So please feel free to improve it.

**Comment by t-b** [comment]
@bourtemb Okay I'll have a look.

File: INSTALL.md (hunk 1 above)
Comment (t-b) [reply INSTALL.md]: Good point. Done.

File: configure/toolchain-i686.cmake (hunk 2 above)
Comment (t-b) [reply configure/toolchain-i686.cmake:1]: @mliszcz Yeah #792 would be a good fit as well, but as I wanted it to be ready for tomorrow I snuck it in here. I've now extracted it into its own commit. Thanks for reminding me.

**Comment by t-b** [comment]
@bourtemb Done. I've moved the wiki stuff into CONTRIBUTING.md as that is the github suggested way of doing things. And I also like it better thant he wiki :)

**Review by mliszcz — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -55,36 +54,10 @@ API reference: [tango-controls.github.io/cppTango-docs](https://tango-controls.g
...
-  we don't want to get worse.
-- Your PR needs two review approvals, including one from the code owners listed [here](https://github.com/tango-controls/cppTango/blob/tango-9-lts/CODEOWNERS).
-- Be prepared to adapt your pull request to the review responses. Code review
-  is done for ensuring higher code quality and communicating implementations
-  details to newcomers and not for annoying anyone or slowing down development.
-
-# Links
-
-[Tango Controls web page](http://tango-controls.org)
+See [CONTRIBUTE file](INSTALL.md).
```
Comment (bourtemb) [inline README.md]:
```suggestion
See [CONTRIBUTING file](CONTRIBUTING.md).
```

**Review by bourtemb — APPROVED** [review]

**Review by mliszcz — APPROVED** [review]

**Comment by t-b** [comment]
@bourtemb @mliszcz Thanks both.

**Pull request merged**